merlin/models/torch/models/base.py

```diff
@@ -0,0 +1,226 @@
+        return self.blocks.values[0]
+
+    def last(self) -> nn.Module:
+        """Returns the last block in the model."""
+        return self.blocks.values[-1]
+
+    def input_schema(self) -> Schema:
+        """Returns the input schema of the model."""
+        if self.schema:
+            return self.schema
+        return Schema([])
+
+
+def initialize(module, data: Union[Dataset, Loader, Batch]):
+    if isinstance(data, (Loader, Dataset)):
+        module.double()  # TODO: Put in data-loader PR to standardize on float-32
+        batch = sample_batch(data, batch_size=1, shuffle=False)
+    elif isinstance(data, Batch):
+        batch = data
+    else:
+        raise RuntimeError(f"Unexpected input type: {type(data)}")
+
+    module.to(batch.device())
+    return module(batch.features, batch=batch)
+
+
+def compute_loss(
+    predictions: Union[torch.Tensor, Dict[str, torch.Tensor]],
+    targets: Optional[Union[torch.Tensor, Dict[str, torch.Tensor]]],
+    model_outputs: Sequence[ModelOutput],
+    compute_metrics: bool = True,
+) -> Dict[str, torch.Tensor]:
+    """Compute the loss and metrics for the given model outputs.
+
+    This function takes in predictions and targets, and a list of model
+    outputs. It computes the loss using the loss function of each model output
+    and averages it. If `compute_metrics` is set to True, it also computes the
+    metrics defined in each model output.
+
+    Parameters
+    ----------
+    predictions: Union[torch.Tensor, Dict[str, torch.Tensor]]
+        The predictions from the model.
+    targets: Optional[Union[torch.Tensor, Dict[str, torch.Tensor]]]
+        The ground truth targets.
+    model_outputs: Sequence[ModelOutput]
+        A list of model outputs. Each model output must have a defined loss
+        function.
+    compute_metrics: bool, optional
+        Whether to compute metrics defined in each model output. Default: True.
+
+    Returns
+    -------
+    Dict[str, torch.Tensor]
+        A dictionary containing the loss and the computed metrics (if any).
+
+    Raises
+    ------
+    RuntimeError: If no model outputs are provided, or if multiple model
+        outputs are provided but only one set of targets is given.
+
+    Example usage
+    -------------
+    >>> predictions = torch.tensor([0.2, 0.3, 0.6, 0.8])
+    >>> targets = torch.tensor([1.0, 0.0, 1.0, 0.0], dtype=torch.float32)
+    >>> binary_output = mm.BinaryOutput(ColumnSchema("target"))
+    >>> results = compute_loss(predictions, targets, [binary_output])
+    >>> results["loss"]
+    tensor(0.7653)
+    >>> results["binary_accuracy"]
+    tensor(0.5000)
+    """
+    if len(model_outputs) < 1:
+        raise RuntimeError("No model outputs found.")
+    if (
+        isinstance(predictions, torch.Tensor)
+        and isinstance(targets, torch.Tensor)
+        and len(model_outputs) > 1
+    ):
+        raise RuntimeError("Multiple outputs but only one target was provided.")
+
```

Comment:
Do we need to check if all `model_outputs` have a `target` property set?

Reply:
I updated the logic to check if `model_outputs` has a `target` when no targets are provided in [lines 205-206](https://github.com/NVIDIA-Merlin/models#diff-ae80586315e21aadf21557b8b7f837710d8a5b5138f3787ca51d68dacd158359R205-R206).